File: BackJoon/P10000_15000/P11000_12000/P11966.c
/*
    [ 문제 ]
    자연수 N이 주어졌을 때, 2의 제곱수면 1을 아니면 0을 출력하는 프로그램을 작성하시오.
*/
#include <stdio.h>

int main(void)
{
    int num = 0;
    scanf("%d", &num);

    if (num == 0)
        printf("0");
    else if ((num & num - 1) == 0)
        printf("1");
    else
        printf("0");
    return 0;
}

File: BackJoon/P10000_15000/P11000_12000/P11966.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Scanner;

public class P11966 {
    public static void main(String[] args) throws IOException {
        P11966 main = new P11966();
        main.run();
    }

    void run() throws IOException {
        WayScanner();
        // WayBufferReader();

    }

    void WayScanner() {
        try (Scanner sc = new Scanner(System.in)) {
            int num = sc.nextInt();
            checkIfPowerOfTwo(num);
        }
    }

    void WayBufferReader() throws IOException {
        try (BufferedReader sc = new BufferedReader(new InputStreamReader(System.in))) {
            String num = sc.readLine();
            checkIfPowerOfTwo(Integer.parseInt(num));
        }
    }

    void checkIfPowerOfTwo(int num) {
        if (num == 0) {
            System.out.println("0");
        } else if ((num & (num - 1)) == 0) {
            System.out.println("1");
        } else {
            System.out.println("0");
        }
    }
}
